session/pessimistic_test.go

```diff
@@ -349,6 +349,37 @@ func (s *testPessimisticSuite) TestBankTransfer(c *C) {
 	tk.MustQuery("select sum(c) from accounts").Check(testkit.Rows("300"))
 }
 
+func (s *testPessimisticSuite) TestLockUnchangedRowKey(c *C) {
+	tk := testkit.NewTestKitWithInit(c, s.store)
+	tk2 := testkit.NewTestKitWithInit(c, s.store)
+	tk.MustExec("drop table if exists unchanged")
+	tk.MustExec("create table unchanged (id int primary key, c int)")
+	tk.MustExec("insert unchanged values (1, 1), (2, 2)")
+
+	tk.MustExec("begin pessimistic")
+	tk.MustExec("update unchanged set c = 1 where id < 2")
+
+	tk2.Se.GetSessionVars().LockWaitTimeout = 1
+	tk2.MustExec("begin pessimistic")
+	err := tk2.ExecToErr("select * from unchanged where id = 1 for update")
```

Comment:
So this time it slows our CI for sure? @coocood 
The `nowait` grammar is not introduced in 3.0? @cfzjywxk

Reply:
No, the `LockWaitTimeout` is set to 1ms, it runs fast.
`nowait` is not picked to 3.0

Reply:
It should be picked.

Reply:
Don't need to pick it for now.